### DISCUSSION TITLE
Seeking example: Render map into an existing webgl context (regl-worldview)

### DISCUSSION THREAD

**Discussion by inactivist** [uber/react-map-gl#1373] [Ideas]
(Not sure if this is the appropriate place to ask this kind of question, please let me know if there's a more appropriate venue.)

I'm looking for examples showing how to render the map inside an existing WebGL context, with other rendered WebGL elements visible above the map.  

In my case, I'm working with an existing app that uses `regl-worldview` to render various elements as children of `WorldView`, so that is something I can't change.  I want to render the map into the XY plane of the Worldview, as if it's a floor texture.  

I see `StaticMap` allows passing an existing context in the `gl` property, and I've played around with that a bit without much luck.  Digging into streetscape.gl, I see that it renders a `StaticMap` as a child of `Core3DViewer -> DeckGL` components, so assume there's a level of compatibility between these two libraries.

Has anyone seen such an example, or if it's a sensible goal?  

In case it's not obvious, I'm a bit of a newbie when it comes to react-map-gl and MapBox.  Same for WebGL (have experience with threejs but that's about it).

**Comment by kylebarron** [comment]
You might be interested in looking at [`@deck.gl/mapbox`](https://deck.gl/docs/api-reference/mapbox/overview) and inspecting the underlying code to see an example of how deck.gl and react-map-gl can share a `gl` context.

**Comment by Pessimistress** [comment]
If you share the WebGLContext then you also need to make sure that the two renderers do not erase the drawing of each other. As far as I know there is no way to block mapbox from wiping the context, so you need to look into if `WorldView` allows you to do so.
